lms/djangoapps/course_goals/handlers.py

```diff
@@ -20,7 +21,7 @@ def update_course_goal_on_enroll_change(sender, instance, **kwargs):  # pylint:
     1) Set the course goal to 'certify' when the user enrolls as a verified user.
     2) Remove the course goal when the user's enrollment is no longer active.
     """
-    course_id = str(instance.course_id).decode('utf8', 'ignore')
+    course_id = six.text_type(instance.course_id)
```

Comment:
I was doing something similar but this is a bit riskier than the change made here: https://github.com/edx/edx-platform/pull/21372/commits/d0f84ab056592e55c7888fa17796c8c644125c76

Reply:
done updated.

Reply:
@feanil  that was causing a test failure on both py2 and py3.